frontend/src/components/Field/YesNo/YesNo.tsx

```diff
@@ -0,0 +1,169 @@
+import { KeyboardEvent, useCallback, useMemo } from 'react'
+import { BiCheck, BiX } from 'react-icons/bi'
+import {
+  Box,
+  forwardRef,
+  HStack,
+  Icon,
+  useRadio,
+  useRadioGroup,
+  UseRadioGroupProps,
+  UseRadioGroupReturn,
+  UseRadioProps,
+  useStyleConfig,
+  VisuallyHidden,
+} from '@chakra-ui/react'
+
+import { YESNO_THEME_KEY } from '~theme/components/Field/YesNo'
+import { FieldColorScheme } from '~theme/foundations/colours'
+
+export interface YesNoProps {
+  /**
+   * Whether YesNo component is disabled.
+   */
+  isDisabled?: boolean
+  /**
+   * Function called once a radio is checked
+   * @param nextValue the value of the checked radio
+   */
+  onChange?: UseRadioGroupProps['onChange']
+  /**
+   * The value of the radio to be `checked`
+   * (in controlled mode)
+   */
+  value?: 'yes' | 'no'
+  /**
+   * The value of the radio to be `checked`
+   * initially (in uncontrolled mode)
+   */
+  defaultValue?: 'yes' | 'no'
+  /**
+   * The `name` attribute forwarded to each `radio` element
+   */
+  name: string
+
+  /**
+   * Color scheme of the component to render. Defaults to `primary`.
+   */
+  colorScheme?: FieldColorScheme
+}
+
+interface YesNoOptionProps extends UseRadioProps {
+  children: React.ReactNode
+
+  /**
+   * Variant of the option for styling to be used for styling.
+   */
+  variant: 'left' | 'right'
+
+  /**
+   * Color scheme of the component to render. Defaults to `primary`.
+   */
+  colorScheme?: FieldColorScheme
+}
+
+const YesNoOption = forwardRef<YesNoOptionProps, 'input'>(
```

Comment:
might be neater to put this in a separate file

Reply:
Done